src/checkers/inference/model/ArithmeticConstraint.java

```diff
@@ -0,0 +1,81 @@
+package checkers.inference.model;
+
+import java.util.Arrays;
+import org.checkerframework.javacutil.ErrorReporter;
+import com.sun.source.tree.Tree.Kind;
+
+/**
+ * Represents a constraint between the result of an arithmetic operation between two operands.
+ * Subclasses of this constraint class denote each specific kind of arithmetic operation, such as
+ * add, sub, mul, div, and mod.
+ *
+ * <p>
+ * This abstract class defines the fields and accessors shared by all arithmetic operations.
+ */
+public abstract class ArithmeticConstraint extends Constraint implements TernaryConstraint {
+
+    private final Slot lefthandside;
+    private final Slot righthandside;
+    private final Slot result;
+
+    protected ArithmeticConstraint(Slot lhs, Slot rhs, Slot result, AnnotationLocation location) {
+        super(Arrays.asList(lhs, rhs, result), location);
+        this.lefthandside = lhs;
+        this.righthandside = rhs;
+        this.result = result;
+    }
+
+    // Alternative getter names
+    public Slot getLHS() {
+        return lefthandside;
+    }
+
+    public Slot getRHS() {
+        return righthandside;
+    }
+
+    public Slot getResult() {
+        return result;
+    }
+
+    @Override
+    public Slot getFirst() {
```

Comment:
To me these three overrides is a bit tricky, as `first` `second`, and `third` doesn't have obvious mapping to `lhs` `rhs` and `result`. Also the functions of three overrides seems duplicated with methods `getLHS` `getRHS` and `getResult`.

Please refer to my another comment on `TernaryConstraint` to see if we can simplify this.

Reply:
If you look at BinaryConstraint, the fields and methods defined go with the name of "first" and "second" but subclasses introduce additional accessor methods relevant to the specific subclass. Eg in SubtypeConstraint there's getSubtype getSupertype. GetFirst() calls getSubtype().

I'm just keeping the architecture consistent between Binary and Ternary constraints. If we think this renaming and reshuffling should change in the future, we can introduce new PRs.

Reply:
Sounds good.